src/test/java/NodeTest.java

```diff
@@ -0,0 +1,46 @@
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class NodeTest {
+    private Node<Integer> classUnderTest;
```

Comment:
Minor, in such tests, I find it a little cleaner to not include a field for the class under test. Many of these tests are inlinable, like `assertNull(new Node<Object>().getData())`. If you need to call a method, cram it in a local variable.